sdk/communication/communication-phone-numbers/test/get.spec.ts

```diff
@@ -5,36 +5,51 @@ import { Recorder, env, isPlaybackMode } from "@azure/test-utils-recorder";
 import { assert } from "chai";
 import { Context } from "mocha";
 import { PhoneNumbersClient } from "../src/phoneNumbersClient";
-import { createRecordedClient } from "./utils/recordedClient";
+import { matrix } from "./utils/matrix";
+import {
+  canCreateRecordedClientWithToken,
+  createRecordedClient,
+  createRecordedClientWithToken
+} from "./utils/recordedClient";
 
-describe("PhoneNumbersClient - get phone number", function() {
-  let recorder: Recorder;
-  let client: PhoneNumbersClient;
+matrix([[true, false]], async function(useAad) {
+  describe(`PhoneNumbersClient - get phone number${useAad ? " [AAD]" : ""}`, function() {
+    let recorder: Recorder;
+    let client: PhoneNumbersClient;
 
-  beforeEach(function(this: Context) {
-    ({ client, recorder } = createRecordedClient(this));
-  });
+    before(function(this: Context) {
+      if (useAad && !canCreateRecordedClientWithToken()) {
+        this.skip();
+      }
+    });
 
-  afterEach(async function(this: Context) {
-    if (!this.currentTest?.isPending()) {
-      await recorder.stop();
-    }
-  });
+    beforeEach(function(this: Context) {
+      ({ client, recorder } = useAad
+        ? createRecordedClientWithToken(this)!
+        : createRecordedClient(this));
+    });
+
+    afterEach(async function(this: Context) {
+      if (!this.currentTest?.isPending()) {
+        await recorder.stop();
+      }
+    });
+
+    it("can get a purchased phone number", async function(this: Context) {
+      const purchasedPhoneNumber = isPlaybackMode() ? "+14155550100" : env.AZURE_PHONE_NUMBER;
+      const { phoneNumber } = await client.getPurchasedPhoneNumber(purchasedPhoneNumber);
+
+      assert.strictEqual(purchasedPhoneNumber, phoneNumber);
+    }).timeout(7000);
```

Comment:
How are these lengths determined btw?

Reply:
Default timeout is 2 seconds. If a test fails because of time out I increase to 5 seconds and rerun. If it still fails I increase by 5 seconds and rerun until I get no failures. I then reduce timeout to 1 second above the slowest run. So for this test case, slowest run was ~6 seconds. 

For LROs, I run them without a time limit first, `timeout(0)`, then set timeout based on how long they take.